src/cli/cli.cpp

```diff
@@ -2236,6 +2237,48 @@ void Interpreter::ProcessService(uint8_t aArgsLength, char *aArgs[])
 }
 #endif
 
+void Interpreter::ProcessNetworkData(uint8_t aArgsLength, char *aArgs[])
+{
+    otError error;
+
+    if (aArgsLength > 2 && strcmp(aArgs[0], "steeringdata") == 0)
+    {
+        if (strcmp(aArgs[1], "has") == 0)
+        {
+            otExtAddress      addr;
+            otJoinerDiscerner discerner;
+
+            discerner.mLength = 0;
+
+            error = Interpreter::ParseJoinerDiscerner(aArgs[2], discerner);
+
+            if (error == OT_ERROR_NOT_FOUND)
+            {
+                VerifyOrExit(Interpreter::Hex2Bin(aArgs[2], addr.m8, sizeof(addr)) == sizeof(addr),
+                             error = OT_ERROR_INVALID_ARGS);
+            }
+            else if (error != OT_ERROR_NONE)
+            {
+                ExitNow();
+            }
+
+            if (discerner.mLength)
+            {
+                ExitNow(error = otNetDataSteeringDataHasJoinerWithDiscerner(mInstance, &discerner));
```

Comment:
We should not treat this as a command error? 
We should better print true/false and prints "Done"?

Reply:
I think an error type gives more information.

Reply:
The information is essential True/False. There is no way we can give more information than that.

I think printing `true/false` along with `Done` is more convenient for CLI parsing (e.g. node.py). We always expect commands to prints `Done` when it finishes successfully, and some command will raise exception when `Error ...' is received. 

`Done` means command executed successfully. `Error ...` means command execution failed. I don't think it is worthy to break this convention here.

Reply:
renamed the api to avoid confusion.